lib/node_modules/@stdlib/math/base/special/gamma-lanczos-sum/src/main.c

```diff
@@ -0,0 +1,128 @@
+
+// BEGIN: rational_pq_p
+
+/**
+* Evaluates a polynomial.
+*
+* ## Notes
+*
+* -   The implementation uses [Horner's rule][horners-method] for efficient computation.
+*
+* [horners-method]: https://en.wikipedia.org/wiki/Horner%27s_method
+*
+* @param x    value at which to evaluate the polynomial
+* @return     evaluated polynomial
+*/
+static double rational_pq_p( const double x ) {
+	return 38474670393.31777 + (x * (36857665043.51951 + (x * (15889202453.72942 + (x * (4059208354.298835 + (x * (680547661.1834733 + (x * (78239755.00312005 + (x * (6246580.776401795 + (x * (341986.3488721347 + (x * (12287.194511824551 + (x * (261.61404416416684 + (x * 2.5066282746310007)))))))))))))))))));
+}
+
+// END: rational_pq_p// BEGIN: rational_pq_p1
+
+/**
+* Evaluates a polynomial.
+*
+* ## Notes
+*
+* -   The implementation uses [Horner's rule][horners-method] for efficient computation.
+*
+* [horners-method]: https://en.wikipedia.org/wiki/Horner%27s_method
+*
+* @param x    value at which to evaluate the polynomial
+* @return     evaluated polynomial
+*/
+static double rational_pq_p1( const double x ) {
+	return 2.5066282746310007 + (x * (261.61404416416684 + (x * (12287.194511824551 + (x * (341986.3488721347 + (x * (6246580.776401795 + (x * (78239755.00312005 + (x * (680547661.1834733 + (x * (4059208354.298835 + (x * (15889202453.72942 + (x * (36857665043.51951 + (x * 38474670393.31777)))))))))))))))))));
+}
+
+// END: rational_pq_p1// BEGIN: rational_pq_q
+
+/**
+* Evaluates a polynomial.
+*
+* ## Notes
+*
+* -   The implementation uses [Horner's rule][horners-method] for efficient computation.
+*
+* [horners-method]: https://en.wikipedia.org/wiki/Horner%27s_method
+*
+* @param x    value at which to evaluate the polynomial
+* @return     evaluated polynomial
+*/
+static double rational_pq_q( const double x ) {
+	return 0.0 + (x * (362880.0 + (x * (1026576.0 + (x * (1172700.0 + (x * (723680.0 + (x * (269325.0 + (x * (63273.0 + (x * (9450.0 + (x * (870.0 + (x * (45.0 + (x * 1.0)))))))))))))))))));
+}
+
+// END: rational_pq_q// BEGIN: rational_pq_q1
+
+/**
+* Evaluates a polynomial.
+*
+* ## Notes
+*
+* -   The implementation uses [Horner's rule][horners-method] for efficient computation.
+*
+* [horners-method]: https://en.wikipedia.org/wiki/Horner%27s_method
+*
+* @param x    value at which to evaluate the polynomial
+* @return     evaluated polynomial
+*/
+static double rational_pq_q1( const double x ) {
+	return 1.0 + (x * (45.0 + (x * (870.0 + (x * (9450.0 + (x * (63273.0 + (x * (269325.0 + (x * (723680.0 + (x * (1172700.0 + (x * (1026576.0 + (x * (362880.0 + (x * 0.0)))))))))))))))))));
+}
+
+// END: rational_pq_q1
+
+/**
+* Calculates the Lanczos sum for the approximation of the gamma function.
+*
+* @param x    input value
+* @return     Lanczos sum approximation
+*
+* @example
+* double v = stdlib_base_gammaLanczosSum( 4.0 );
+* // returns ~950.366
+*/
+double stdlib_base_gamma_lanczos_sum( const double x ) {
+	double ax;
```

Comment:
This will also need to be updated once `evalrational-compile-c` is used.

Reply:
@gunjjoshi Was this function auto-generated by the script or did you manually copy? I'm a little bit surprised that the script understood to use the constant macro at L43.

Reply:
Hey @kgryte, the function is indeed auto-generated. But the only modification I did was to replace the macro. Is there any other way for handling macros here ?

Reply:
Manually editing is risky, as, if we ever update the constants and rerun the script, your manual edits will be overwritten. My thinking is that we need to update https://github.com/stdlib-js/stdlib/blob/develop/lib/node_modules/%40stdlib/math/base/tools/evalrational-compile-c/lib/main.js in order to handle this particular edge case. Namely, if the "ratio" is infinity, then inline `1.0 / 0.0` (or `-1.0 / 0.0`).

Reply:
In particular, we need to update `value2string` https://github.com/stdlib-js/stdlib/blob/develop/lib/node_modules/%40stdlib/math/base/tools/evalrational-compile-c/lib/main.js#L53. Would you like to open a separate PR for this? Would also need to add associated tests.

Reply:
Yes @kgryte, I would like to open a separate PR regarding this, making necessary changes in the `value2string` function.

Reply:
Great! Thanks!

Reply:
Now since we changed `evalrational-compile-c`, I ran the script again, to auto-generate the function. It auto-generated the functions. It passes all the tests and benchmarks well.

But, while pushing the changes, it gives the following error:

```bash
lib/node_modules/@stdlib/random/base/minstd-shuffle/README.md
  78:1-87:4  error  9:1   error   Displayed return value is `20739838`, but expected `1421600654` instead   stdlib/doctest  eslint  remark-lint

✖ 1 error
make: *** [markdown-examples-javascript-files] Error 1

Encountered an error when running README JavaScript examples.

error: failed to push some refs to 'https://github.com/gunjjoshi/stdlib'
```

The [package](https://github.com/stdlib-js/stdlib/blob/cf29ed0fc4f7108b5be01daca89ef72e6ebfcc1f/lib/node_modules/%40stdlib/random/base/minstd-shuffle/README.md?plain=1#L86) that leads to the error is very much unrelated to what we are doing here. I wonder what might be the reason for this.

cc: @kgryte

Reply:
This happened when attempting to rebase, I am guessing?

Reply:
For this one time, you can circumvent Git hooks by appending `--no-verify` when running the `push` command.

Reply:
I also addressed the lint error. If you rebase on the latest `develop`, you should be able to push as you would normally.

Reply:
*sigh* Looks like I should have recommended just merging in the latest develop, rather than `rebase`. No worries. Looks like GitHub is still picking up on the changes in this PR and we'll squash-merge so all good.

Reply:
Yes, it happened while rebasing. But after getting the latest develop, it works well. Thanks !
Also, in the latest commit pushed, I've regenerated the auto-generated function, and it now works well with infinities, as we intended.